src/AccountService/AccountService.API/Controllers/DepartmentController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using AccountService.Application.Commands.Departments;
+using AccountService.Application.Commands.Users;
+using AccountService.Application.Models.Departments;
+using AccountService.Application.Models.Users;
+using AccountService.Application.Queries.Departments;
+using AccountService.Application.Queries.User;
+using CustomHelper.Authentication.Attributes;
+using CustomHelper.Authentication.Enums;
+using CustomHelper.Exception;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AccountService.API.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class DepartmentController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+        public DepartmentController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetDepartmentById(Ulid Id, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var department = await _mediator.Send(new GetDepartmentQuery(Id), cancellationToken);
+
+                if (department == null)
+                {
+                    return NotFound(Id);
+                }
+
+                return Ok(department);
+            }
+            catch
+            {
+                throw;
```

Comment:
Return BadRequest in case of exceptions

Reply:
For this purpose, I have middleware for handling exceptions, which returns 500 code and the error